File: UKParliament.CodeTest.Data/HATEOAS/Pagination.cs
namespace UKParliament.CodeTest.Data.HATEOAS;

public class Pagination
{
    public int Total { get; set; }
    public int PerPage { get; set; }
    public int CurrentPage { get; set; }
    public int FinalPage { get; set; }
    public string? FirstPageUrl { get; set; }
    public string? FinalPageUrl { get; set; }
    public string? NextPageUrl { get; set; }
    public string? PrevPageUrl { get; set; }
    public string? Path { get; set; }
    public int From { get; set; }
    public int To { get; set; }
}

File: UKParliament.CodeTest.Web/ClientApp/src/app/models/pagination.ts
export interface Pagination {
    total: number,
    perPage: number,
    currentPage: number,
    finalPage: number,
    firstPageUrl?: string,
    finalPageUrl?: string,
    nextPageUrl?: string,
    prevPageUrl?: string,
    path?: string,
    from: number,
    to: number,
}